Error while processing route: vault.index in WebUI

Getting the following Javascript error while trying to access the WebUI in Chrome Version 76.0.3809.100 on OSX.

Vault Version: 1.2.2

Error while processing route: vault.index Cannot read property 'includes' of null TypeError: Cannot read property 'includes' of null

Console image attached. I have restarted the browser and performed a hard reload/clear of the cache. 

Seems to be working fine in Firefox.

<img width="1219" alt="Screenshot of Google Chrome (29-8-19, 8-50-02 am)" src="https://user-images.githubusercontent.com/723789/63902275-4ae65100-ca3b-11e9-93bb-9f787d4f817e.png">

I am having the same issue. Using the authentication method "Username". Using version: Vault 1.3.0. If I login with incognito mode in chrome (or clear my storage) then it is fine I am able to login.

The error log from chrome:
`vendor-1decb7c3f4690ab9d322972e8db32da0.js:2127 Error while processing route: vault.index Cannot read property 'includes' of null TypeError: Cannot read property 'includes' of null
    at https://dvolsvault000/ui/assets/vault-eb665d5ecf606896894349c5cf2805d8.js:2243:15
    at Array.forEach (<anonymous>)
    at n.checkForRootToken (https://dvolsvault000/ui/assets/vault-eb665d5ecf606896894349c5cf2805d8.js:2242:65)
    at n.init (https://dvolsvault000/ui/assets/vault-eb665d5ecf606896894349c5cf2805d8.js:2212:242)
    at n [as init] (https://dvolsvault000/ui/assets/vendor-1decb7c3f4690ab9d322972e8db32da0.js:2460:9)
    at m (https://dvolsvault000/ui/assets/vendor-1decb7c3f4690ab9d322972e8db32da0.js:2369:218)
    at Function.e.create (https://dvolsvault000/ui/assets/vendor-1decb7c3f4690ab9d322972e8db32da0.js:2378:8)
    at e.n.create (https://dvolsvault000/ui/assets/vendor-1decb7c3f4690ab9d322972e8db32da0.js:881:18)
    at https://dvolsvault000/ui/assets/vendor-1decb7c3f4690ab9d322972e8db32da0.js:855:76
    at l (https://dvolsvault000/ui/assets/vendor-1decb7c3f4690ab9d322972e8db32da0.js:861:45)
(anonymous) @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:2127
error @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:2127
O @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:2137
c.triggerEvent @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:2069
t.trigger @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4306
c.transitionDidError @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:2069
(anonymous) @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4297
g @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4499
O @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4509
M @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4507
w @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4505
t.invoke @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4041
t.flush @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4033
t.flush @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4046
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
vendor-1decb7c3f4690ab9d322972e8db32da0.js:2224 Uncaught TypeError: Cannot read property 'includes' of null
    at vault-eb665d5ecf606896894349c5cf2805d8.js:2243
    at Array.forEach (<anonymous>)
    at n.checkForRootToken (vault-eb665d5ecf606896894349c5cf2805d8.js:2242)
    at n.init (vault-eb665d5ecf606896894349c5cf2805d8.js:2212)
    at n [as init] (vendor-1decb7c3f4690ab9d322972e8db32da0.js:2460)
    at m (vendor-1decb7c3f4690ab9d322972e8db32da0.js:2369)
    at Function.e.create (vendor-1decb7c3f4690ab9d322972e8db32da0.js:2378)
    at e.n.create (vendor-1decb7c3f4690ab9d322972e8db32da0.js:881)
    at vendor-1decb7c3f4690ab9d322972e8db32da0.js:855
    at l (vendor-1decb7c3f4690ab9d322972e8db32da0.js:861)
(anonymous) @ vault-eb665d5ecf606896894349c5cf2805d8.js:2243
checkForRootToken @ vault-eb665d5ecf606896894349c5cf2805d8.js:2242
init @ vault-eb665d5ecf606896894349c5cf2805d8.js:2212
n @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:2460
m @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:2369
e.create @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:2378
n.create @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:881
(anonymous) @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:855
l @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:861
r.lookup @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:846
lookup @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:2306
bt @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:1825
o.get @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:1713
re @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:1615
ie @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:1618
re @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:1613
get @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:2323
addHeaders @ vault-eb665d5ecf606896894349c5cf2805d8.js:3
_preRequest @ vault-eb665d5ecf606896894349c5cf2805d8.js:6
ajax @ vault-eb665d5ecf606896894349c5cf2805d8.js:8
n @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:2460
health @ vault-eb665d5ecf606896894349c5cf2805d8.js:62
(anonymous) @ vault-eb665d5ecf606896894349c5cf2805d8.js:2415
u @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:757
(anonymous) @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:755
e.<computed> @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:764
_resumeGenerator @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:7738
_handleResolvedContinueValue @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:7748
_proceed @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:7741
_scheduleProceed @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:7741
_start @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:7724
_startTaskInstance @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:7707
_flushQueues @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:7704
schedule @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:7700
_performShared @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:7780
perform @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:7779
fetchVersion @ vault-eb665d5ecf606896894349c5cf2805d8.js:2423
beforeModel @ vault-eb665d5ecf606896894349c5cf2805d8.js:1698
t.runBeforeModelHook @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4334
(anonymous) @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4324
g @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4499
O @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4509
M @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4507
t.invoke @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4041
t.flush @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4033
t.flush @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4046
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
characterData (async)
t @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4017
n._scheduleAutorun @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4121
n._end @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4108
_boundAutorunEnd @ vendor-1decb7c3f4690ab9d322972e8db32da0.js:4058
Show 18 more frames
0:26 Service Worker registration succeeded. Scope is https://dvolsvault000/v1/sys/storage/raft/snapshot
`

--- a/vault/external_tests/policy/policy_test.go
+++ b/vault/external_tests/policy/policy_test.go
@@ -185,69 +185,134 @@ func TestPolicy_NoConfiguredPolicy(t *testing.T) {
 }
 
 func TestPolicy_TokenRenewal(t *testing.T) {
-	coreConfig := &vault.CoreConfig{
-		CredentialBackends: map[string]logical.Factory{
-			"userpass": credUserpass.Factory,
+	cases := []struct {
+		name             string
+		tokenPolicies    []string
+		identityPolicies []string
+	}{
+		{
+			"default only",
+			nil,
+			nil,
+		},
+		{
+			"with token policies",
+			[]string{"token-policy"},
+			nil,
+		},
+		{
+			"with identity policies",
+			nil,
+			[]string{"identity-policy"},
+		},
+		{
+			"with token and identity policies",
+			[]string{"token-policy"},
+			[]string{"identity-policy"},
 		},
 	}
-	cluster := vault.NewTestCluster(t, coreConfig, &vault.TestClusterOptions{
-		HandlerFunc: vaulthttp.Handler,
-	})
-	cluster.Start()
-	defer cluster.Cleanup()
 
-	core := cluster.Cores[0].Core
-	vault.TestWaitActive(t, core)
-	client := cluster.Cores[0].Client
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			coreConfig := &vault.CoreConfig{
+				CredentialBackends: map[string]logical.Factory{
+					"userpass": credUserpass.Factory,
+				},
+			}
+			cluster := vault.NewTestCluster(t, coreConfig, &vault.TestClusterOptions{
+				HandlerFunc: vaulthttp.Handler,
+			})
+			cluster.Start()
+			defer cluster.Cleanup()
 
-	// Enable userpass auth
-	err := client.Sys().EnableAuthWithOptions("userpass", &api.EnableAuthOptions{
-		Type: "userpass",
-	})
-	if err != nil {
-		t.Fatal(err)
-	}
+			core := cluster.Cores[0].Core
+			vault.TestWaitActive(t, core)
+			client := cluster.Cores[0].Client
 
-	// Add a user to userpass backend
-	_, err = client.Logical().Write("auth/userpass/users/testuser", map[string]interface{}{
-		"password": "testpassword",
-	})
-	if err != nil {
-		t.Fatal(err)
-	}
+			// Enable userpass auth
+			err := client.Sys().EnableAuthWithOptions("userpass", &api.EnableAuthOptions{
+				Type: "userpass",
+			})
+			if err != nil {
+				t.Fatal(err)
+			}
 
-	// Authenticate
-	secret, err := client.Logical().Write("auth/userpass/login/testuser", map[string]interface{}{
-		"password": "testpassword",
-	})
-	if err != nil {
-		t.Fatal(err)
-	}
-	clientToken := secret.Auth.ClientToken
+			// Add a user to userpass backend
+			data := map[string]interface{}{
+				"password": "testpassword",
+			}
+			if len(tc.tokenPolicies) > 0 {
+				data["token_policies"] = tc.tokenPolicies
+			}
+			_, err = client.Logical().Write("auth/userpass/users/testuser", data)
+			if err != nil {
+				t.Fatal(err)
+			}
 
-	// Verify the policy exists in the login response
-	if !strutil.EquivalentSlices(secret.Auth.TokenPolicies, []string{"default"}) {
-		t.Fatalf("policy mismatch, got token policies: %v", secret.Auth.TokenPolicies)
-	}
+			// Set up entity if we're testing against an identity_policies
+			if len(tc.identityPolicies) > 0 {
+				auths, err := client.Sys().ListAuth()
+				if err != nil {
+					t.Fatal(err)
+				}
+				userpassAccessor := auths["userpass/"].Accessor
 
-	if !strutil.EquivalentSlices(secret.Auth.Policies, []string{"default"}) {
-		t.Fatalf("policy mismatch, got token policies: %v", secret.Auth.Policies)
-	}
+				resp, err := client.Logical().Write("identity/entity", map[string]interface{}{
+					"name":     "test-entity",
+					"policies": tc.identityPolicies,
+				})
+				if err != nil {
+					t.Fatal(err)
+				}
+				entityID := resp.Data["id"].(string)
 
-	// Renew token
-	secret, err = client.Logical().Write("auth/token/renew", map[string]interface{}{
-		"token": clientToken,
-	})
-	if err != nil {
-		t.Fatal(err)
-	}
+				// Create an alias
+				resp, err = client.Logical().Write("identity/entity-alias", map[string]interface{}{
+					"name":           "testuser",
+					"mount_accessor": userpassAccessor,
+					"canonical_id":   entityID,
+				})
+				if err != nil {
+					t.Fatal(err)
+				}
+			}
 
-	// Verify the policy exists in the renewal response
-	if !strutil.EquivalentSlices(secret.Auth.TokenPolicies, []string{"default"}) {
-		t.Fatalf("policy mismatch, got token policies: %v", secret.Auth.TokenPolicies)
-	}
+			// Authenticate
+			secret, err := client.Logical().Write("auth/userpass/login/testuser", map[string]interface{}{
+				"password": "testpassword",
+			})
+			if err != nil {
+				t.Fatal(err)
+			}
+			clientToken := secret.Auth.ClientToken
 
-	if !strutil.EquivalentSlices(secret.Auth.Policies, []string{"default"}) {
-		t.Fatalf("policy mismatch, got token policies: %v", secret.Auth.Policies)
+			// Verify the policies exist in the login response
+			expectedTokenPolicies := append([]string{"default"}, tc.tokenPolicies...)
+			if !strutil.EquivalentSlices(secret.Auth.TokenPolicies, expectedTokenPolicies) {
+				t.Fatalf("policy mismatch:\nexpected: %v\ngot: %v", expectedTokenPolicies, secret.Auth.TokenPolicies)
+			}
+
+			expectedPolicies := append(expectedTokenPolicies, tc.identityPolicies...)
+			if !strutil.EquivalentSlices(secret.Auth.Policies, expectedPolicies) {
+				t.Fatalf("policy mismatch:\nexpected: %v\ngot: %v", expectedPolicies, secret.Auth.Policies)
+			}
+
+			// Renew token
+			secret, err = client.Logical().Write("auth/token/renew", map[string]interface{}{
+				"token": clientToken,
+			})
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			// Verify the policies exist in the renewal response
+			if !strutil.EquivalentSlices(secret.Auth.TokenPolicies, expectedTokenPolicies) {
+				t.Fatalf("policy mismatch:\nexpected: %v\ngot: %v", expectedTokenPolicies, secret.Auth.TokenPolicies)
+			}
+
+			if !strutil.EquivalentSlices(secret.Auth.Policies, expectedPolicies) {
+				t.Fatalf("policy mismatch:\nexpected: %v\ngot: %v", expectedPolicies, secret.Auth.Policies)
+			}
+		})
 	}
 }

--- a/vault/external_tests/policy/policy_test.go
+++ b/vault/external_tests/policy/policy_test.go
@@ -8,8 +8,10 @@ import (
 	"github.com/hashicorp/go-hclog"
 	"github.com/hashicorp/vault/api"
 	"github.com/hashicorp/vault/builtin/credential/ldap"
+	credUserpass "github.com/hashicorp/vault/builtin/credential/userpass"
 	ldaphelper "github.com/hashicorp/vault/helper/testhelpers/ldap"
 	vaulthttp "github.com/hashicorp/vault/http"
+	"github.com/hashicorp/vault/sdk/helper/strutil"
 	"github.com/hashicorp/vault/sdk/logical"
 	"github.com/hashicorp/vault/vault"
 )
@@ -179,5 +181,73 @@ func TestPolicy_NoConfiguredPolicy(t *testing.T) {
 	// Verify that the lease renewal extended the duration properly.
 	if float64(secret.Auth.LeaseDuration) < (1 * time.Hour).Seconds() {
 		t.Fatalf("failed to renew lease, got: %v", secret.Auth.LeaseDuration)
+	}
+}
+
+func TestPolicy_TokenRenewal(t *testing.T) {
+	coreConfig := &vault.CoreConfig{
+		CredentialBackends: map[string]logical.Factory{
+			"userpass": credUserpass.Factory,
+		},
+	}
+	cluster := vault.NewTestCluster(t, coreConfig, &vault.TestClusterOptions{
+		HandlerFunc: vaulthttp.Handler,
+	})
+	cluster.Start()
+	defer cluster.Cleanup()
+
+	core := cluster.Cores[0].Core
+	vault.TestWaitActive(t, core)
+	client := cluster.Cores[0].Client
+
+	// Enable userpass auth
+	err := client.Sys().EnableAuthWithOptions("userpass", &api.EnableAuthOptions{
+		Type: "userpass",
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Add a user to userpass backend
+	_, err = client.Logical().Write("auth/userpass/users/testuser", map[string]interface{}{
+		"password": "testpassword",
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Authenticate
+	secret, err := client.Logical().Write("auth/userpass/login/testuser", map[string]interface{}{
+		"password": "testpassword",
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	clientToken := secret.Auth.ClientToken
+
+	// Verify the policy exists in the login response
+	if !strutil.EquivalentSlices(secret.Auth.TokenPolicies, []string{"default"}) {
+		t.Fatalf("policy mismatch, got token policies: %v", secret.Auth.TokenPolicies)
+	}
+
+	if !strutil.EquivalentSlices(secret.Auth.Policies, []string{"default"}) {
+		t.Fatalf("policy mismatch, got token policies: %v", secret.Auth.Policies)
+	}
+
+	// Renew token
+	secret, err = client.Logical().Write("auth/token/renew", map[string]interface{}{
+		"token": clientToken,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Verify the policy exists in the renewal response
+	if !strutil.EquivalentSlices(secret.Auth.TokenPolicies, []string{"default"}) {
+		t.Fatalf("policy mismatch, got token policies: %v", secret.Auth.TokenPolicies)
+	}
+
+	if !strutil.EquivalentSlices(secret.Auth.Policies, []string{"default"}) {
+		t.Fatalf("policy mismatch, got token policies: %v", secret.Auth.Policies)
 	}
 }

--- a/vault/external_tests/policy/policy_test.go
+++ b/vault/external_tests/policy/policy_test.go
@@ -289,7 +289,11 @@ func TestPolicy_TokenRenewal(t *testing.T) {
 			// Verify the policies exist in the login response
 			expectedTokenPolicies := append([]string{"default"}, tc.tokenPolicies...)
 			if !strutil.EquivalentSlices(secret.Auth.TokenPolicies, expectedTokenPolicies) {
-				t.Fatalf("policy mismatch:\nexpected: %v\ngot: %v", expectedTokenPolicies, secret.Auth.TokenPolicies)
+				t.Fatalf("token policy mismatch:\nexpected: %v\ngot: %v", expectedTokenPolicies, secret.Auth.TokenPolicies)
+			}
+
+			if !strutil.EquivalentSlices(secret.Auth.IdentityPolicies, tc.identityPolicies) {
+				t.Fatalf("identity policy mismatch:\nexpected: %v\ngot: %v", tc.identityPolicies, secret.Auth.IdentityPolicies)
 			}
 
 			expectedPolicies := append(expectedTokenPolicies, tc.identityPolicies...)
@@ -308,6 +312,10 @@ func TestPolicy_TokenRenewal(t *testing.T) {
 			// Verify the policies exist in the renewal response
 			if !strutil.EquivalentSlices(secret.Auth.TokenPolicies, expectedTokenPolicies) {
 				t.Fatalf("policy mismatch:\nexpected: %v\ngot: %v", expectedTokenPolicies, secret.Auth.TokenPolicies)
+			}
+
+			if !strutil.EquivalentSlices(secret.Auth.IdentityPolicies, tc.identityPolicies) {
+				t.Fatalf("identity policy mismatch:\nexpected: %v\ngot: %v", tc.identityPolicies, secret.Auth.IdentityPolicies)
 			}
 
 			if !strutil.EquivalentSlices(secret.Auth.Policies, expectedPolicies) {

--- a/vault/request_handling.go
+++ b/vault/request_handling.go
@@ -836,9 +836,8 @@ func (c *Core) handleRequest(ctx context.Context, req *logical.Request) (retResp
 	}
 
 	// Only the token store is allowed to return an auth block, for any
-	// other request this is an internal error. We exclude renewal of a token,
-	// since it does not need to be re-registered
-	if resp != nil && resp.Auth != nil && !strings.HasPrefix(req.Path, "auth/token/renew") {
+	// other request this is an internal error.
+	if resp != nil && resp.Auth != nil {
 		if !strings.HasPrefix(req.Path, "auth/token/") {
 			c.logger.Error("unexpected Auth response for non-token backend", "request_path", req.Path)
 			retErr = multierror.Append(retErr, ErrInternalError)
@@ -868,24 +867,34 @@ func (c *Core) handleRequest(ctx context.Context, req *logical.Request) (retResp
 			return nil, nil, ErrInternalError
 		}
 
-		resp.Auth.TokenPolicies = policyutil.SanitizePolicies(resp.Auth.Policies, policyutil.DoNotAddDefaultPolicy)
-		switch resp.Auth.TokenType {
-		case logical.TokenTypeBatch:
-		case logical.TokenTypeService:
-			if err := c.expiration.RegisterAuth(ctx, &logical.TokenEntry{
-				TTL:         auth.TTL,
-				Policies:    auth.TokenPolicies,
-				Path:        resp.Auth.CreationPath,
-				NamespaceID: ns.ID,
-			}, resp.Auth); err != nil {
-				// Best-effort clean up on error, so we log the cleanup error as
-				// a warning but still return as internal error.
-				if err := c.tokenStore.revokeOrphan(ctx, resp.Auth.ClientToken); err != nil {
-					c.logger.Warn("failed to clean up token lease during auth/token/ request", "request_path", req.Path, "error", err)
+		// We skip expiration manager registration for token renewal since it
+		// does not need to be re-registered
+		if strings.HasPrefix(req.Path, "auth/token/renew") {
+			// We build the "policies" list to be returned by starting with
+			// token policies, and add identity policies right after this
+			// conditional
+			resp.Auth.Policies = policyutil.SanitizePolicies(resp.Auth.TokenPolicies, policyutil.DoNotAddDefaultPolicy)
+		} else {
+			resp.Auth.TokenPolicies = policyutil.SanitizePolicies(resp.Auth.Policies, policyutil.DoNotAddDefaultPolicy)
+
+			switch resp.Auth.TokenType {
+			case logical.TokenTypeBatch:
+			case logical.TokenTypeService:
+				if err := c.expiration.RegisterAuth(ctx, &logical.TokenEntry{
+					TTL:         auth.TTL,
+					Policies:    auth.TokenPolicies,
+					Path:        resp.Auth.CreationPath,
+					NamespaceID: ns.ID,
+				}, resp.Auth); err != nil {
+					// Best-effort clean up on error, so we log the cleanup error as
+					// a warning but still return as internal error.
+					if err := c.tokenStore.revokeOrphan(ctx, resp.Auth.ClientToken); err != nil {
+						c.logger.Warn("failed to clean up token lease during auth/token/ request", "request_path", req.Path, "error", err)
+					}
+					c.logger.Error("failed to register token lease during auth/token/ request", "request_path", req.Path, "error", err)
+					retErr = multierror.Append(retErr, ErrInternalError)
+					return nil, auth, retErr
 				}
-				c.logger.Error("failed to register token lease during auth/token/ request", "request_path", req.Path, "error", err)
-				retErr = multierror.Append(retErr, ErrInternalError)
-				return nil, auth, retErr
 			}
 		}
 

